File: ExtraWork/diaryapp/entry.py
class Entry:
    def __init__(self, id, title, content,date):
        if title is None or title.strip() == "":
            raise ValueError("title cannot be null or empty")
        if content is None or content.strip() == "":
            raise ValueError("content cannot be null or empty")
        if id < 0:
            raise ValueError("id cannot be negative")

        self.__id = id
        self.__title = title
        self.__content = content
        self.__date = date

    @property
    def id(self):
        return self.__id

    @property
    def title(self):
        return self.__title

    @title.setter
    def title(self, title):
        if title is None or title.strip() == "":
            raise ValueError("title cannot be null or empty")
        self.__title = title

    @property
    def content(self):
        return self.__content

    @content.setter
    def content(self, content):
        if content is None or content.strip() == "":
            raise ValueError("content cannot be null or empty")
        self.__content = content


    @property
    def date(self):
        return self.__date

    @date.setter
    def date(self, date):
        if date is None or date.strip() == "":
            raise ValueError("date cannot be null or empty")
        self.__date = date

    def update(self, title, content,date):
        if title and title.strip():
            self.title = title
        if content and content.strip():
            self.content = content

File: src/diaryApp/Entry.java
package diaryApp;

import java.text.SimpleDateFormat;
import java.util.Date;

public class Entry {
    private int id;
    private String title;
    private String content;
    private Date date;

    public Entry(int id, String title, String content) {
        if (title == null || title.isEmpty()) {
            throw new IllegalArgumentException("title cannot be null or empty");
        }
        if (id < 0) {
            throw new IllegalArgumentException("id cannot be negative");
        }
        this.id = id;
        this.title = title;
        this.content = content;
        this.date = new Date();
    }

    public void setTitle(String title) {
        if (title == null || title.isEmpty()) {
            throw new IllegalArgumentException("title cannot be null or empty");
        } else {
            this.title = title;
        }
    }

    public void setContent(String content) {
        if (content == null || content.isEmpty()) {
            throw new IllegalArgumentException("content cannot be null or empty");
        } else {
            this.content = content;
        }
    }

    public int getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public String getContent() {
        return content;
    }

    public String getDate() {
        SimpleDateFormat dateFormat = new SimpleDateFormat("dd/MM/yyyy");
        return dateFormat.format(date);
    }

    public void update(String title, String content) {
        if (title != null && !title.isEmpty()) {
            this.title = title;
        }
        if (content != null && !content.isEmpty()) {
            this.content = content;
        }
    }
}